ripley/externals/data_loch.py

```diff
@@ -205,13 +205,24 @@ def get_edo_enrollment_updates(since_timestamp):
     return safe_execute_rds(sql, **params)
 
 
-def get_grades_with_demographics(term_id, section_ids):
+def get_grades_with_demographics(term_id, section_ids, instructor_uid):
     params = {
         'earliest_term_id': str(app.config['CANVAS_OLDEST_OFFICIAL_TERM']),
         'section_ids': section_ids,
         'term_id': term_id,
     }
-    sql = """WITH course AS (
+    exclude_other_instructors_sections = ''  # noqa
+    if instructor_uid:
+        params['instructor_uid'] = instructor_uid
+        exclude_other_instructors_sections = """AND sec.instructor_uid = c.instructor_uid
+            AND NOT EXISTS (
+                SELECT * FROM sis_data.edo_sections sec
+                WHERE sec.instructor_uid <> %(instructor_uid)s
+                AND sec.sis_term_id = enr.sis_term_id
+                AND sec.sis_section_id = enr.sis_section_id
+            )
```

Comment:
@lyttam Do we have any concerns about receiving back no grade data in the (fairly frequent) case of co-taught primary sections? Or are we confident that we're not going to have any co-taught courses in our first batch?

It seems to me that anyone who's an instructor of record on a graded section should be able to see those grades, regardless of whether additional instructors of record exist. But I don't know the full scope of requirements that have been set for us.

Reply:
@lyttam When we were asked for instructor siloing the requirement shouldn’t restrict tool deployment only to single instructor course sites. The above query seems to suggest that?

 We are to allow courses that are co-taught by multiple instructors.  When we calculate grade averages and aggregates we should roll up all sections where the instructor is teaching(listed as primary or otherwise). As long as they are listed as an instructor in the section we should include data from those sis sections.

Reply:
Thanks @sandeep-jay. If I'm reading the query right, I think that means we can do without this `AND NOT EXISTS` condition.

Reply:
That makes sense @pauline2k.

Reply:
Thanks @pauline2k and @sandeep-jay, I agree. Amended.